src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -215,11 +217,68 @@ public void canReplaceFeedInfo() throws SQLException, IOException {
         );
     }
 
+    @Test
+    public void canPreserveCustomFieldsInStops() throws IOException {
+        String stops = generateStopsWithCustomFields();
+        FeedTransformation transformation = PreserveCustomFieldsTransformation.create(stops, "stops");
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        targetVersion = createFeedVersion(
+                feedSource,
+                zipFolderFiles("fake-agency-with-only-calendar-dates")
+        );
+        LOG.info("Checking assertions.");
+        assertEquals(
+                2,
+                targetVersion.feedTransformResult.tableTransformResults.get(0).customColumnsAdded,
+                "stops.txt custom column count should equal input csv data # of custom columns"
+        );
+
+        assertEquals(
+                2,
+                targetVersion.feedTransformResult.tableTransformResults.get(0).updatedCount,
+                "stops.txt row count modified with custom content should equal input csv data # of custom columns"
+        );
+    }
+
+    @Test
+    public void canAddCustomFile() throws IOException {
+        String customCsv = generateCustomCsvData();
+        FeedTransformation transformation = AddCustomFileFromStringTransformation.create(customCsv, "custom-file");
+        FeedTransformRules transformRules = new FeedTransformRules(transformation);
+        feedSource.transformRules.add(transformRules);
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+        targetVersion = createFeedVersion(
+                feedSource,
+                zipFolderFiles("fake-agency-with-only-calendar-dates")
+        );
+
+        LOG.info("Checking assertions.");
+        assertEquals(
+                2,
+                targetVersion.feedTransformResult.tableTransformResults.get(0).addedCount,
+                "custom-file.txt custom row count should equal input csv data # of rows"
+        );
+
+    }
+
     private static String generateFeedInfo(String feedId) {
         // Add feed_info csv data (purposefully with two rows, even though this is not valid GTFS).
         return String.format(
             "feed_id,feed_publisher_name,feed_publisher_url,feed_lang\n%s,BART,https://www.bart.gov/,en\n2,abc,https://example.com",
             feedId
         );
     }
+    private static String generateStopsWithCustomFields() {
+        return "stop_id, custom_column1, custom_column2"
+            + "\n4u6g, customValue1, customValue2"
+            + "\n1234567, customValue1, customValue2";
```

Comment:
Nit. When reviewing the output it would be clearer if these values where `customValue3` and `customValue4` so that I know the correct values have been written to the correct row.

Reply:
Done in https://github.com/ibi-group/datatools-server/pull/536/commits/19face8619d29a6c495202c37fa708fe65508ce1